### PR TITLE
ruby: init ruby sub-project, implement kafka publisher

### PR DESCRIPTION
**Pull request by mxmCherry**

### CODE REVIEW

File: ruby/bps-kafka/bps-kafka.gemspec [hunk 1]
```diff
@@ -0,0 +1,22 @@
+Gem::Specification.new do |s|
+  s.name        = 'bps-kafka'
+  s.version     = File.read(File.expand_path('../.version', __dir__)).strip
+  s.platform    = Gem::Platform::RUBY
+
+  s.licenses    = ['Apache-2.0']
+  s.summary     = 'Kafka adapter for bps'
+  s.description = 'https://github.com/bsm/bps'
+
+  s.authors     = ['Dimitrij Denissenko']
```
Comment (dim) [inline ruby/bps-kafka/bps-kafka.gemspec:10]: feel free to mention yourself :)

File: ruby/bps-kafka/spec/bps/kafka_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  Kafka.new(kafka_addrs).each_message(topic: topic_name).take(num_messages)
+end
+
+RSpec.describe BPS::Kafka::ReliablePublisher, if: run_spec do
+  subject do
+    described_class.new(kafka_addrs)
+  end
+
+  it_behaves_like 'publisher',
+                  read_messages: read_messages,
```
Comment (dim) [inline ruby/bps-kafka/spec/bps/kafka_spec.rb:23]: a simpler way is to add an abstract `def read_messages` method to the shared behaviour and overload here

File: ruby/bps/bps.gemspec [hunk 3]
```diff
@@ -0,0 +1,19 @@
+Gem::Specification.new do |s|
+  s.name        = 'bps'
+  s.version     = File.read(File.expand_path('../.version', __dir__)).strip
+  s.platform    = Gem::Platform::RUBY
+
+  s.licenses    = ['Apache-2.0']
+  s.summary     = 'Multi-platform pubsub adapter'
+  s.description = 'Minimalist abstraction for publish-subscribe'
+
+  s.authors     = ['Dimitrij Denissenko']
```
Comment (dim) [inline ruby/bps/bps.gemspec:10]: I did nothing here :)

File: ruby/bps-kafka/lib/bps/kafka.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      def close
+        @client.close
+        nil
+      end
+    end
+  end
+
+  register_publisher('kafka') do |url, opts|
+    addrs = CGI.unescape(url.host).split(',')
+    Kafka::ReliablePublisher.new(addrs, **opts)
```
Comment (dim) [inline ruby/bps-kafka/lib/bps/kafka.rb:40]: a note for future PR(s):
this should be configurable via URL param, i.e. if we use a reliable vs async publisher, i.e. `mode=sync`

File: ruby/bps-kafka/lib/bps/kafka.rb [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+          @topic_name = topic_name
+        end
+
+        def publish(msg_data)
+          @client.deliver_message(msg_data, topic: @topic_name, retries: DELIVER_MESSAGE_RETRIES)
+          nil
+        end
+      end
+
+      def initialize(seed_brokers, client_id: nil)
```
Comment (dim) [inline ruby/bps-kafka/lib/bps/kafka.rb:23]: a note for future PR(s): we should make retries configurable, i.e.: `, retries: DELIVER_MESSAGE_RETRIES` and store as instance var.

File: ruby/bps-kafka/bps-kafka.gemspec (hunk 1 above)
Comment (mxmCherry) [reply ruby/bps-kafka/bps-kafka.gemspec:10]: I'd rather mention the whole team (I suppose, this `authors` thingy is kinda like "who to reach for support"?)

File: ruby/bps-kafka/lib/bps/kafka.rb (hunk 4 above)
Comment (mxmCherry) [reply ruby/bps-kafka/lib/bps/kafka.rb:40]: I think it's better to do what's done in Go counterpart: `kafka://` vs `kafka+sync://` (multi-scheme registration)

Comment (mxmCherry) [reply ruby/bps-kafka/lib/bps/kafka.rb:40]: I'll keep the `ReliablePublisher` (I don't like this name much, but it's not a SyncProducer, I think as `kafka://` (default impl). At least, for now.

**Pull request merged**